src/server/transaction.h

```diff
@@ -124,6 +124,11 @@ class Transaction {
   void LogJournalOnShard(EngineShard* shard, journal::Entry::Payload&& payload,
                          uint32_t shard_cnt) const;
 
+  // In some cases for non auto-journaling commands we want to enable the auto journal flow.
+  void SetAutoJournal() {
+    no_auto_enable_auto_journal_ = true;
+  }
+
```

Comment:
Important detail here is that this implies we have only one shard

I'd maybe add this as as command or DCHECK this

Reply:
I am not sure this must be the case of reanable the auto journal. True that its the current use right now but not must and I dont think we need to enforce it. 
The condition to use this can be the number of shards we run on but it also can be the param sent in the command.
f.e set with PXAT could use this flow. yes set is single shard command but if it was mset this would be valid as well

Reply:
> if it was mset this would be valid as well

We can run only on one shard because this is an unguarded field that might be accessed form multiple threads in the case of mset, so it would need to be an atomic